pkg/timelock/timelock.go

```diff
@@ -166,22 +180,36 @@ func (tw *Worker) Listen() error {
 	shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second*5)
 	defer cancel()
 
-	<-isclosed.All(shutdownCtx, schedulingDone, historyDone, newDone, processingDone)
+	<-isclosed.All(shutdownCtx, schedulingDone, historyDone, newDone, processingDone) //nolint:contextcheck
 
 	return nil
 }
 
 // setupFilterQuery returns an ethereum.FilterQuery initialized to watch the Timelock contract.
-func (tw *Worker) setupFilterQuery() ethereum.FilterQuery {
+func (tw *Worker) setupFilterQuery(fromBlock *big.Int) ethereum.FilterQuery {
 	return ethereum.FilterQuery{
 		Addresses: tw.address,
-		FromBlock: tw.fromBlock,
+		FromBlock: fromBlock,
+	}
+}
+
+// retrieveNewLogs returns a "control channel" and a "logs channels". The logs channel is where
+// new log events will be asynchronously pushed to.
+//
+// The actual retrieveal is performed by either `subscribeNewLogs`, if the node connection
```

Comment:
```suggestion
// The actual retrieval is performed by either `subscribeNewLogs`, if the node connection
```

Reply:
Thanks for caching this. I'll fix it and add to a future PR just to save the additional approval. We're a bit late on this task.